drivers/hwinfo/hwinfo_mcux_src_rev2.c

```diff
@@ -0,0 +1,80 @@
+/*
+ * Copyright (c) 2022 NXP
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#define DT_DRV_COMPAT nxp_imx_src_rev2
+
+#include <soc.h>
+#include <drivers/hwinfo.h>
+#include <string.h>
+#include <sys/byteorder.h>
+#include <fsl_soc_src.h>
+
+BUILD_ASSERT(DT_NUM_INST_STATUS_OKAY(DT_DRV_COMPAT) == 1, "No nxp,imx-src compatible device found");
+
+int z_impl_hwinfo_get_reset_cause(uint32_t *cause)
+{
+	uint32_t flags = 0;
+	uint32_t reason = SRC_GetResetStatusFlags((SRC_Type *)DT_INST_REG_ADDR(0));
+
+	if (reason & (kSRC_M7CoreIppUserResetFlag | kSRC_M4CoreIppUserResetFlag)) {
```

Comment:
Are the reset cause for both cores always the same?
If not, how does the user know which cause applies to which core?

Reply:
The reset status flags for each core should be the same-this is a good point. I will change the driver to use one flag or the other based on which core is being targeted during compilation.